src/code/z_fireobj.c

```diff
@@ -101,39 +95,41 @@ void FireObj_UpdateStateTransitions(PlayState* play, FireObj* fire) {
     Vec3f dist;
 
     FireObj_StepSize(fire);
-    fire->unk26++;
+    fire->timer++;
     if (fire->ignitionDelay > 0) {
         fire->ignitionDelay--;
     } else if (fire->ignitionDelay == 0) {
         fire->ignitionDelay = -1;
-        if ((fire->state == FIRE_STATE_3) || (fire->state == FIRE_STATE_1)) {
-            nextState = FIRE_STATE_0;
+        if ((fire->state == FIRE_STATE_NOT_LIT) || (fire->state == FIRE_STATE_SHRINKING)) {
+            nextState = FIRE_STATE_GROWING;
         } else {
-            nextState = FIRE_STATE_1;
+            nextState = FIRE_STATE_SHRINKING;
         }
         FireObj_SetState(fire, fire->dynamicSizeStep, nextState);
     }
-    if ((fire->flags & 1) && (fire->state != FIRE_STATE_3) &&
+
+    if ((fire->flags & FIRE_FLAG_1) && (fire->state != FIRE_STATE_NOT_LIT) &&
         WaterBox_GetSurface1_2(play, &play->colCtx, fire->position.x, fire->position.z, &waterY, &waterBox) &&
         (waterY - fire->position.y > 6500.0f * fire->yScale)) {
-        FireObj_SetState(fire, fire->dynamicSizeStep, FIRE_STATE_3);
+        FireObj_SetState(fire, fire->dynamicSizeStep, FIRE_STATE_NOT_LIT);
```

Comment:
FIRE_FLAG_1 seems to be the only version that checks for water to be extinguishable, can we name it? Not sure what the second flag is tho.

Reply:
The second flag seems to be like "can interact with stick"
So what about `FIRE_FLAG_WATER_EXTINGUISHABLE` and `FIRE_FLAG_INTERACT_STICK`

Reply:
Bump, this PR is good for me once these are named

Reply:
Whoops, completely forgot to check this PR again :sweat_smile: